test/jest/victory-scatter/victory-scatter.test.js

```diff
@@ -0,0 +1,250 @@
+            { x: 5, y: 5 }
+          ]}
+          {...svgDimensions}
+        />
+      );
+      const domain = { x: [0, 5], y: [0, 5] };
+
+      const points = container.querySelectorAll("path");
+      const svgCoordinates = Array.from(points).map(getSvgPointCoordinates);
+      const coordinates = svgCoordinates.map((coord) => {
+        return convertSvgCoordinatesToCartesian(coord, svgDimensions, domain);
+      });
+
+      expect(coordinates).toEqual([
+        [0, 0],
+        [2, 3],
+        [5, 5]
+      ]);
+    });
+
+    it("does not render data with null x or y values", () => {
+      const data = [
+        { x: 1, y: 2 },
+        { x: null, y: 4 },
+        { x: 5, y: null }
+      ];
+      const { container } = render(<VictoryScatter data={data} />);
+      expect(container.querySelectorAll("path")).toHaveLength(1);
+    });
+  });
+
+  describe("event handling", () => {
+    it("attaches an event to the parent svg", () => {
+      const clickHandler = jest.fn();
+      const { container } = render(
+        <VictoryScatter
+          events={[
+            {
+              target: "parent",
+              eventHandlers: { onClick: clickHandler }
+            }
+          ]}
+        />
+      );
+      const svg = container.querySelector("svg");
+      fireEvent.click(svg);
+      expect(clickHandler).toBeCalled();
+      // the first argument is the standard evt object
+      expect(Object.keys(clickHandler.mock.calls[0][1])).toEqual(
+        expect.arrayContaining(["data", "scale", "width", "height", "style"])
+      );
+    });
+
+    it("attaches an event to data", () => {
+      const clickHandler = jest.fn();
+      const { container } = render(
+        <VictoryScatter
+          events={[
+            {
+              target: "data",
+              eventHandlers: { onClick: clickHandler }
+            }
+          ]}
+        />
+      );
+      const Data = container.querySelectorAll("path");
+      Data.forEach((node, index) => {
+        fireEvent.click(node);
+        expect(clickHandler).toHaveBeenCalled();
+        expect(`${clickHandler.mock.calls[index][2]}`).toEqual(`${index}`);
+      });
+    });
+
+    it("attaches an event to a label", () => {
+      const clickHandler = jest.fn();
+      const data = [
+        { eventKey: 0, _x: 0, _y: 0, x: 0, y: 0, label: "0" },
+        { eventKey: 1, _x: 1, _y: 1, x: 1, y: 1, label: "1" },
+        { eventKey: 2, _x: 2, _y: 2, x: 2, y: 2, label: "2" }
+      ];
+      render(
+        <VictoryScatter
+          data={data}
+          labelComponent={<VictoryLabel data-testid="label" />}
+          events={[
+            {
+              target: "labels",
+              eventHandlers: { onClick: clickHandler }
+            }
+          ]}
+        />
+      );
+      const Labels = screen.getAllByTestId("label");
+      Labels.forEach((node, index) => {
+        fireEvent.click(node);
+        expect(clickHandler).toHaveBeenCalled();
```

Comment:
I'm not sure that these `toHaveBeenCalled()` expectations in the loops are necessary because since they're not being reset it's really only asserting for the first node, and the following `mock.calls[index]` implicitly asserts that it was called the number of times expected.

Reply:
Good point; I recommend adding `clickHandler.mockClear()` to the top of the loop

Reply:
I'll add a `beforeEach` and `afterEach` to this describe block that does this for me.